magicctapipe/scripts/lst1_magic/semi_automatic_scripts/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,740 @@
+            "cat list_dl0_ok.txt | while read line\n",
+            "do\n",
+            f"    cd {target_dir}/{source_name}/../\n",
+            f"    conda run -n {env_name} lst1_magic_mc_dl0_to_dl1 --input-file $line --output-dir {target_dir}/{source_name}/DL1/MC/{particle_type}/$SAMPLE --config-file {target_dir}/{source_name}/config_DL0_to_DL1.yaml --focal_length_choice {focal_length}>>$LOG 2>&1\n\n",
+            "done\n",
+            "",
+        ]
+        f.writelines(lines_of_config_file)
+
+
+def lists_and_bash_gen_MAGIC(
+    target_dir, telescope_ids, MAGIC_runs, source, env_name, NSB_match
+):
+
+    """
+    Below we create a bash script that links the the MAGIC data paths to each subdirectory.
+
+    Parameters
+    ----------
+    target_dir : str
+        Directory to store the results
+    telescope_ids : list
+        List of the telescope IDs (set by the user)
+    MAGIC_runs : array
+        MAGIC dates and runs to be processed
+    source : str
+        Name of the target
+    env_name : str
+        Name of the environment
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+    process_name = source
+    lines = [
+        "#!/bin/sh\n\n",
+        "#SBATCH -p short\n",
+        f"#SBATCH -J {process_name}\n",
+        "#SBATCH -n 1\n\n",
+        f"#SBATCH --output={target_dir}/v{__version__}/{source}/DL1/slurm-linkMAGIC-%x.%j.out"
+        f"#SBATCH --error={target_dir}/v{__version__}/{source}/DL1/slurm-linkMAGIC-%x.%j.err"
+        "ulimit -l unlimited\n",
+        "ulimit -s unlimited\n",
+        "ulimit -a\n",
+    ]
+    with open(f"{source}_linking_MAGIC_data_paths.sh", "w") as f:
+        f.writelines(lines)
+        if NSB_match:
+
+            if (len(MAGIC_runs) == 2) and (len(MAGIC_runs[0]) == 10):
+                MAGIC = MAGIC_runs
+
+                MAGIC_runs = []
+                MAGIC_runs.append(MAGIC)
+
+            for i in MAGIC_runs:
+
+                if telescope_ids[-1] > 0:
+                    lines = [
+                        f'export IN1=/fefs/onsite/common/MAGIC/data/M2/event/Calibrated/{i[0].split("_")[0]}/{i[0].split("_")[1]}/{i[0].split("_")[2]}\n',
+                        f"export OUT1={target_dir}/v{__version__}/{source}/DL1/M2/{i[0]}/{i[1]}/logs \n",
+                        f"ls $IN1/*{i[1][-2:]}.*_Y_*.root > $OUT1/list_dl0.txt\n",
+                    ]
+                    f.writelines(lines)
+
+                f.write("\n")
+                if telescope_ids[-2] > 0:
+                    lines = [
+                        f'export IN1=/fefs/onsite/common/MAGIC/data/M1/event/Calibrated/{i[0].split("_")[0]}/{i[0].split("_")[1]}/{i[0].split("_")[2]}\n',
+                        f"export OUT1={target_dir}/v{__version__}/{source}/DL1/M1/{i[0]}/{i[1]}/logs \n",
+                        f"ls $IN1/*{i[1][-2:]}.*_Y_*.root > $OUT1/list_dl0.txt\n",
+                    ]
+                    f.writelines(lines)
+        else:
```

Comment:
the lines for NSB_MATCH False look the same as above, or am I missing something?
they can be extracted from the if to make the code shorter